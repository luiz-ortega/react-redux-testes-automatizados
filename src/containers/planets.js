import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchPlanets } from '../store/actions';
import { userPlanetList } from '../store/selectors/planets-selectors';
import Filters from './filters';
import Deck from '../components/planets/deck/main';

export default function Planets() {
  const dispatch = useDispatch();
  const { loading, error } = useSelector(store => store.planets);
  const list = useSelector(userPlanetList);

  useEffect(() => {
    dispatch(fetchPlanets());
  }, [dispatch]);

  return (
    <>
      {error ? <h4>So sorry...</h4> : null}
      {loading ? <h4>Loading...</h4> : null}
      {!loading ? (
        <>
          <div className="mb-5">
            <Filters />
          </div>
          <Deck dataSource={list} />
        </>
      ) : null}
    </>
  );
}
