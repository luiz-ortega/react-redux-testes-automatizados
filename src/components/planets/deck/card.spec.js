import React from 'react';
import { render } from '@testing-library/react';
import Card from './card';
import planets from '../../../../test/stubs/planets';

describe('Component: Card', () => {
  function renderComponent(overrides = {}) {
    return render(<Card planet={planets.results[0]} />);
  }

  describe('Displays all elements', () => {
    it('should match snapshot', () => {
      const { baseElement } = renderComponent();
      expect(baseElement).toMatchSnapshot();
    });
  });
});
