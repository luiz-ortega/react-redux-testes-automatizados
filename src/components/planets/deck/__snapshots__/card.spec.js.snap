// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Card Displays all elements should match snapshot 1`] = `
<body>
  <div>
    <li
      class="col-span-1 flex flex-col text-center bg-gray-100 rounded-lg shadow divide-y divide-gray-200"
    >
      <div
        class="flex-1 flex flex-col p-8"
      >
        <img
          alt="Tatooine"
          class="w-32 h-32 flex-shrink-0 mx-auto bg-black rounded-full"
          title="Tatooine"
        />
        <div
          class="mt-8"
        >
          <div
            class="bg-white shadow overflow-hidden sm:rounded-lg"
          >
            <div
              class="px-4 py-5 sm:px-6"
            >
              <h3
                class="text-lg leading-6 font-medium text-gray-900"
              >
                Tatooine
              </h3>
              <p
                class="mt-1 max-w-2xl text-sm text-gray-500"
              >
                /api/planets/1/
              </p>
            </div>
            <div
              class="border-t border-gray-200 px-4 py-5 sm:px-6"
            >
              <dl
                class="grid grid-cols-1 gap-x-4 gap-y-8 sm:grid-cols-2"
              >
                <div
                  class="sm:col-span-1"
                >
                  <dt
                    class="text-sm font-medium text-gray-500"
                  >
                    Climate
                  </dt>
                  <dd
                    class="mt-1 text-sm text-gray-900"
                  >
                    arid
                  </dd>
                </div>
                <div
                  class="sm:col-span-1"
                >
                  <dt
                    class="text-sm font-medium text-gray-500"
                  >
                    Terrain
                  </dt>
                  <dd
                    class="mt-1 text-sm text-gray-900"
                  >
                    desert
                  </dd>
                </div>
                <div
                  class="sm:col-span-1"
                >
                  <dt
                    class="text-sm font-medium text-gray-500"
                  >
                    Gravity
                  </dt>
                  <dd
                    class="mt-1 text-sm text-gray-900"
                  >
                    1 standard
                  </dd>
                </div>
                <div
                  class="sm:col-span-1"
                >
                  <dt
                    class="text-sm font-medium text-gray-500"
                  >
                    Surface water
                  </dt>
                  <dd
                    class="mt-1 text-sm text-gray-900"
                  >
                    1
                  </dd>
                </div>
                <div
                  class="sm:col-span-1"
                >
                  <dt
                    class="text-sm font-medium text-gray-500"
                  >
                    Diameter
                  </dt>
                  <dd
                    class="mt-1 text-sm text-gray-900"
                  >
                    10465
                  </dd>
                </div>
                <div
                  class="sm:col-span-1"
                >
                  <dt
                    class="text-sm font-medium text-gray-500"
                  >
                    Population
                  </dt>
                  <dd
                    class="mt-1 text-sm text-gray-900"
                  >
                    200000
                  </dd>
                </div>
                <div
                  class="sm:col-span-2"
                >
                  <dt
                    class="text-sm font-medium text-gray-500"
                  >
                    About
                  </dt>
                  <dd
                    class="mt-1 text-sm text-gray-900"
                  />
                </div>
                <div
                  class="sm:col-span-2"
                >
                  <dt
                    class="text-sm font-medium text-gray-500"
                  >
                    Films
                  </dt>
                  <dd
                    class="mt-1 text-sm text-gray-900"
                  >
                    <ul
                      class="border border-gray-200 rounded-md divide-y divide-gray-200"
                    >
                      <li
                        class="pl-3 pr-4 py-3 flex items-center justify-between text-sm"
                      >
                        <div
                          class="w-0 flex-1 flex items-center"
                        >
                          <svg
                            class="h-5 w-5 text-gray-400"
                            fill="none"
                            stroke="currentColor"
                            viewBox="0 0 24 24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M7 4v16M17 4v16M3 8h4m10 0h4M3 12h18M3 16h4m10 0h4M4 20h16a1 1 0 001-1V5a1 1 0 00-1-1H4a1 1 0 00-1 1v14a1 1 0 001 1z"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="2"
                            />
                          </svg>
                          <span
                            class="ml-2 flex-1 w-0 truncate"
                          >
                            /api/films/1/
                          </span>
                        </div>
                        <div
                          class="ml-4 flex-shrink-0"
                        >
                          <a
                            class="font-medium text-indigo-600 hover:text-indigo-500"
                            href="/api/films/1/"
                          >
                            Visit
                          </a>
                        </div>
                      </li>
                      <li
                        class="pl-3 pr-4 py-3 flex items-center justify-between text-sm"
                      >
                        <div
                          class="w-0 flex-1 flex items-center"
                        >
                          <svg
                            class="h-5 w-5 text-gray-400"
                            fill="none"
                            stroke="currentColor"
                            viewBox="0 0 24 24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M7 4v16M17 4v16M3 8h4m10 0h4M3 12h18M3 16h4m10 0h4M4 20h16a1 1 0 001-1V5a1 1 0 00-1-1H4a1 1 0 00-1 1v14a1 1 0 001 1z"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="2"
                            />
                          </svg>
                          <span
                            class="ml-2 flex-1 w-0 truncate"
                          >
                            /api/films/3/
                          </span>
                        </div>
                        <div
                          class="ml-4 flex-shrink-0"
                        >
                          <a
                            class="font-medium text-indigo-600 hover:text-indigo-500"
                            href="/api/films/3/"
                          >
                            Visit
                          </a>
                        </div>
                      </li>
                      <li
                        class="pl-3 pr-4 py-3 flex items-center justify-between text-sm"
                      >
                        <div
                          class="w-0 flex-1 flex items-center"
                        >
                          <svg
                            class="h-5 w-5 text-gray-400"
                            fill="none"
                            stroke="currentColor"
                            viewBox="0 0 24 24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M7 4v16M17 4v16M3 8h4m10 0h4M3 12h18M3 16h4m10 0h4M4 20h16a1 1 0 001-1V5a1 1 0 00-1-1H4a1 1 0 00-1 1v14a1 1 0 001 1z"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="2"
                            />
                          </svg>
                          <span
                            class="ml-2 flex-1 w-0 truncate"
                          >
                            /api/films/4/
                          </span>
                        </div>
                        <div
                          class="ml-4 flex-shrink-0"
                        >
                          <a
                            class="font-medium text-indigo-600 hover:text-indigo-500"
                            href="/api/films/4/"
                          >
                            Visit
                          </a>
                        </div>
                      </li>
                      <li
                        class="pl-3 pr-4 py-3 flex items-center justify-between text-sm"
                      >
                        <div
                          class="w-0 flex-1 flex items-center"
                        >
                          <svg
                            class="h-5 w-5 text-gray-400"
                            fill="none"
                            stroke="currentColor"
                            viewBox="0 0 24 24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M7 4v16M17 4v16M3 8h4m10 0h4M3 12h18M3 16h4m10 0h4M4 20h16a1 1 0 001-1V5a1 1 0 00-1-1H4a1 1 0 00-1 1v14a1 1 0 001 1z"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="2"
                            />
                          </svg>
                          <span
                            class="ml-2 flex-1 w-0 truncate"
                          >
                            /api/films/5/
                          </span>
                        </div>
                        <div
                          class="ml-4 flex-shrink-0"
                        >
                          <a
                            class="font-medium text-indigo-600 hover:text-indigo-500"
                            href="/api/films/5/"
                          >
                            Visit
                          </a>
                        </div>
                      </li>
                      <li
                        class="pl-3 pr-4 py-3 flex items-center justify-between text-sm"
                      >
                        <div
                          class="w-0 flex-1 flex items-center"
                        >
                          <svg
                            class="h-5 w-5 text-gray-400"
                            fill="none"
                            stroke="currentColor"
                            viewBox="0 0 24 24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M7 4v16M17 4v16M3 8h4m10 0h4M3 12h18M3 16h4m10 0h4M4 20h16a1 1 0 001-1V5a1 1 0 00-1-1H4a1 1 0 00-1 1v14a1 1 0 001 1z"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="2"
                            />
                          </svg>
                          <span
                            class="ml-2 flex-1 w-0 truncate"
                          >
                            /api/films/6/
                          </span>
                        </div>
                        <div
                          class="ml-4 flex-shrink-0"
                        >
                          <a
                            class="font-medium text-indigo-600 hover:text-indigo-500"
                            href="/api/films/6/"
                          >
                            Visit
                          </a>
                        </div>
                      </li>
                    </ul>
                  </dd>
                </div>
              </dl>
            </div>
          </div>
        </div>
      </div>
    </li>
  </div>
</body>
`;
